### ISSUE TITLE
Glitches in an interactive shell spawned via 'podman exec', but not 'podman run', due to the presence of OPOST in the outer PTY and missing ONLCR in the inner PTY

### ISSUE THREAD

**Issue by Zlopez** [debarshiray/toolbox#14]
## Description of issue
I tried to use vim in tmux in fedora-toolbox container and I have this weird rendering issue.

When I add newline to script, every line that should be indented by smarttab option is replacing line numbers, but in the buffer it is still on the same place.
This is how it looks before I add a new line:
![screenshot from 2018-11-09 09-47-35](https://user-images.githubusercontent.com/6943409/48252532-519ccc80-e405-11e8-82c7-c46a6049319b.png)
And this is how it looks after:
![screenshot from 2018-11-09 09-47-46](https://user-images.githubusercontent.com/6943409/48252542-59f50780-e405-11e8-9d43-487c436d1fcf.png)

This issue is only happening in tmux, I didn't saw this when the vim is started outside of tmux.

## Versions of components
vim-common-8.1.511-1.fc29.x86_64
tmux-2.8-1.fc29.x86_64
fedora-toolbox commit cf2347fe562dd78210c7245f5ded268d8ea7f18a

## How to reproduce
1. ./fedora-toolbox create
2. ./fedora-toolbox enter
3. sudo dnf install tmux vim
4. tmux
5. vim (some python file)
6. :set number
7. :set smarttab
8. Add new line to file somewhere in the code

**Comment by debarshiray** [comment]
I have heard about glitches in Emacs, but I don't think tmux was involved and I don't know if it's the same problem as the one you have reported here. Needs debugging.

**Comment by victortoso** [comment]
If I run fedora-toolbox enter > tmux > tig, I can reproduce glitches very easily by browsing commits;
If I run tmux > fedora-toolbox > tig, I can't reproduce glitches (or they don't come so easily)

Let me know how can I help, no real knowledge on how to debug this.

**Comment by mildred** [comment]
I had vim and less rendering issues (I was not using tmux) with fedora-toolbox, and I found out it was due to locale settings not being set properly within the toolbox.

Just run `locale` within the toolbox, and see if there are any errors showing. Installing a locale package fixed the problem in my case. See #60

**Comment by Frederik-Baetens** [comment]
Hmm, i have similar issues, where output from ls will still partially remain in vim. [example (look for the tablr and ui on the 5th line](https://imgur.com/GcnctVE)

**Comment by Frederik-Baetens** [comment]
@debarshiray should i make a separate issue about this? Since my issue isn't fixed by installing a locale package.

**Comment by debarshiray** [comment]
Did you try `dnf -y swap glibc-minimal-langpack glibc-all-langpacks` from #60?

**Comment by Frederik-Baetens** [comment]
I have tried that now, and that doesn't fix my issue, so i think that my issue is a different one from this.

**Comment by Frederik-Baetens** [comment]
Can you reproduce this? Just open a tmux in toolbox, split vertically, type ls a few times, and then type clear, you will see this glitch.

**Comment by kamalmarhubi** [comment]
I had opened [a related issue against tmux](https://github.com/tmux/tmux/issues/1745), but after further experimentation I think this is more likely a toolbox/podman issue. My repro is very simple:

Steps to reproduce:
- start tmux with no config inside a toolbox container
- produce any output, eg `seq 10`
- create a new vertical pane

At this point, some output from the `seq 10` command disappears.

**Comment by debarshiray** [comment]
Ok, thanks for continuing to dig into it and following up!

I am not very surprised of the possibility of this being a Toolbox or Podman problem. However, I don't understand what that problem is. It's likely that somebody with a deeper understanding of terminals will be able to figure this out much easier than me.

**Comment by debarshiray** [comment]
Looks like there's a difference between the different ways in which one can get an interactive shell inside a container with Podman.

This doesn't display the glitch when splitting horizontally with `ctrl+b %` in tmux:
```
$ podman run --env TERM=$TERM --rm -it registry.fedoraproject.org/f30/fedora-toolbox:30 /bin/bash
```

This is what Toolbox uses, and it does show the glitch:
```
$ podman create --name foo registry.fedoraproject.org/f30/fedora-toolbox:30 sleep +Inf
$ podman start foo
$ podman exec --env TERM=$TERM -it foo /bin/bash
```

**Comment by debarshiray** [comment]
Filed https://github.com/containers/libpod/issues/3179

**Comment by mildred** [comment]
I can reproduce this on recent toolboxes with no tmux (just neovim). The langpack fix I found earlier does not work any more as the locale is no longer shared between the host and the toolbox (I get the default `en_US.UTF-8` locale)

**Comment by debarshiray** [comment]
This is now fixed in Podman 1.5.0.
